config-builder/src/main/java/com/ca/apim/gateway/cagatewayconfig/bundle/builder/EncassEntityBuilder.java

```diff
@@ -168,7 +170,7 @@ private Entity buildEncassEntity(AnnotatedBundle annotatedBundle, Bundle bundle,
         buildAndAppendPropertiesElement(properties, document, encassAssertionElement);
         Entity entity = getEntityWithNameMapping(ENCAPSULATED_ASSERTION_TYPE, name, encassName, id, encassAssertionElement, guid);
 
-        if (isRedeployableBundle || !(encass.isReusable() || isAnnotatedEntity(encass, annotatedEntity))) {
+        if (isRedeployableBundle || !(isReusable || isAnnotatedEntity(encass, annotatedEntity))) {
```

Comment:
@uppoju , do we still need isRedeployableBundle , user can make a choice based on isReusable annotation

Reply:
yes, isRedeployableBundle is not required
@graju256 , what's your opinion ?

Reply:
I don't think @redeployable annotation becomes obsolete because of this change.  Scope of @reusable annotation is changed but that doesn't mean that we never find bundles reusing the entities. 

Two scenarios:
1. **@bundle** --> it never shares anything with any other bundles. Hence, redeployable is implicit and no need to use @redeployable annotation. 
2. **@bundle + @reusable** --> There's a possibility that other bundle might share some of the entities with it. Hence, the bundle is not treated as redeployable. So, in this case, one can override this mechanism using @redeplyable annotation. 

I didn't really go through the full chnges of this PR. Please let me know if my understanding is wrong.

Reply:
@graju256 , for 2 point, instead of overriding behaviour with @redeployable we can remove @reusable annotation right.
Admin can decide whether he wants reusable or redeployable

Reply:
I think you are correct. if not reusable, we will see it like redeployable.